$OpenBSD: patch-Source_kwsys_SystemTools_cxx,v 1.10 2011/10/14 17:22:10 dcoppa Exp $

Don't spam systrace by attempting to create dirs which already exist.

--- Source/kwsys/SystemTools.cxx.orig	Tue Oct  4 18:09:25 2011
+++ Source/kwsys/SystemTools.cxx	Fri Oct  7 09:15:20 2011
@@ -227,11 +227,13 @@ inline void Realpath(const char *path, kwsys_stl::stri
 }
 #else
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
 inline int Mkdir(const char* dir)
 {
-  return mkdir(dir, 00777);
+  struct stat sb;
+  return stat(dir, &sb) && mkdir(dir, 00777);
 }
 inline int Rmdir(const char* dir)
 {
